upstash/client/client.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/imroc/req"
+	"github.com/upstash/terraform-provider-upstash/upstash/utils"
+)
+
+const UPSTASH_API_ENDPOINT = "https://api.upstash.com"
+
+type UpstashClient struct {
+	Email  string
+	Apikey string
+}
+
+func NewUpstashClient(email string, apikey string) *UpstashClient {
+	return &UpstashClient{
+		Email:  email,
+		Apikey: apikey,
+	}
+}
+
+func (c *UpstashClient) DeleteCalls(endpointExtension string, body interface{}, errMessage string) (err error) {
+	resp, err := req.Delete(
+		UPSTASH_API_ENDPOINT+endpointExtension,
+		req.Header{"Accept": "application/json"},
+		req.Header{"Authorization": utils.BasicAuth(c.Email, c.Apikey)},
+		req.BodyJSON(body),
+	)
+	if err != nil {
+		return err
+	}
+	if resp.Response().StatusCode != http.StatusOK && resp.Response().StatusCode != http.StatusAccepted {
+		return errors.New(errMessage + " failed, status code: " + strconv.Itoa(resp.Response().StatusCode) + " response: " + resp.String())
+	}
+	return err
+}
+
+func (c *UpstashClient) GetCalls(endpointExtension string, errMessage string) (response *req.Resp, err error) {
+	resp, err := req.Get(
+		UPSTASH_API_ENDPOINT+endpointExtension,
+		req.Header{"Accept": "application/json"},
+		req.Header{"Authorization": utils.BasicAuth(c.Email, c.Apikey)},
+	)
+	if err != nil {
+		return resp, err
```

Comment:
we are returning the response in this one(in case of an error) but not in PostCalls method. It might be better to have a consistency here.